systems/src/apu/electronic_control_box.rs

```diff
@@ -0,0 +1,263 @@
+                    || self.apu_n.get::<percent>() > 99.5))
+                || (self.turbine_state != TurbineState::Starting
+                    && self.apu_n.get::<percent>() > 95.))
+    }
+
+    pub fn get_egt_warning_temperature(&self) -> ThermodynamicTemperature {
+        self.egt_warning_temperature
+    }
+
+    pub fn get_egt_caution_temperature(&self) -> ThermodynamicTemperature {
+        const WARNING_TO_CAUTION_DIFFERENCE: f64 = 33.;
+        ThermodynamicTemperature::new::<degree_celsius>(
+            self.egt_warning_temperature.get::<degree_celsius>() - WARNING_TO_CAUTION_DIFFERENCE,
+        )
+    }
+
+    pub fn get_egt(&self) -> ThermodynamicTemperature {
+        self.egt
+    }
+
+    pub fn get_n(&self) -> Ratio {
+        self.apu_n
+    }
+}
+impl ApuStartContactorController for ElectronicControlBox {
+    /// Indicates if the APU start contactor should be closed.
+    fn should_close_start_contactor(&self) -> bool {
+        if self.is_inoperable() {
+            false
+        } else {
+            match self.turbine_state {
+                TurbineState::Shutdown => {
+                    self.master_is_on && self.start_is_on && self.air_intake_flap_fully_open
+                }
+                TurbineState::Starting => self.apu_n.get::<percent>() < 55.,
+                _ => false,
+            }
+        }
+    }
+}
+impl AirIntakeFlapController for ElectronicControlBox {
+    /// Indicates if the air intake flap should be opened.
+    fn should_open_air_intake_flap(&self) -> bool {
+        match self.turbine_state {
+            TurbineState::Shutdown => self.master_is_on,
+            TurbineState::Starting => true,
+            // While running, the air intake flap remains open.
+            TurbineState::Running => true,
+            // Manual shutdown sequence: the air intake flap closes at N = 7%.
+            TurbineState::Stopping => self.master_is_on || 7. <= self.apu_n.get::<percent>(),
+        }
+    }
+}
+impl TurbineController for ElectronicControlBox {
+    /// Indicates if the start sequence should be started.
+    fn should_start(&self) -> bool {
+        self.start_contactor_is_energized
+    }
+
+    fn should_stop(&self) -> bool {
+        self.is_auto_shutdown()
+            || self.is_emergency_shutdown()
+            || (!self.master_is_on
+                && self.turbine_state != TurbineState::Starting
+                && !self.bleed_air_valve_was_open_in_last(Duration::from_millis(
+                    ElectronicControlBox::BLEED_AIR_COOLDOWN_DURATION_MILLIS,
+                )))
+    }
+}
+impl BleedAirValveController for ElectronicControlBox {
+    fn should_open_bleed_air_valve(&self) -> bool {
+        self.fault != Some(ApuFault::ApuFire)
+            && self.master_is_on
+            && self.apu_n.get::<percent>() > 95.
+            && self.bleed_is_on
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+enum ApuFault {
+    ApuFire,
+    FuelLowPressure,
```

Comment:
Are starter faults a thing? I guess we'll only need those once we have a fault system, or alternatively if the electrical system cannot provide enough power to ignite the turbine.

Reply:
Of course it might be that the batteries do not have enough power to finish the start. We can implement this once we have a functioning ELEC system.